source/gui/__init__.py

```diff
@@ -483,49 +484,90 @@ def onInstallCommand(self, evt):
-		except:  # noqa: E722
+		except WindowsError as e:
+			# 1223 is "The operation was canceled by the user."
+			if e.winerror == 1223:
+				# Same as if the user selected "no" in the initial dialog.
+				log.debug("Run of COM Registration Fixing Tool canceled during UAC.")
+				return
+			else:
+				log.error("Could not execute fixCOMRegistrations command", exc_info=True)
+				error = e  # Hold for later display to the user
+				return  # Safe because of finally block
+		except Exception:
 			log.error("Could not execute fixCOMRegistrations command", exc_info=True)
-		progressDialog.done()
-		del progressDialog
+			return
+		finally:  # Clean up the progress dialog, and display any important error to the user before returning
+			progressDialog.done()
+			del progressDialog
+			self.postPopup()
+			# If there was a Windows error, inform the user because it may have support value
+			if error is not None:
+				messageBox(
+					_(
+						# Translators: message shown to the user on COM Registration Fix fail
+						"The COM Registration Fixing Tool was unsuccessful. This Windows "
+						"error may provide more information.    {}",
+					).format(error),
+					# Translators: Added to the title of the dialog showing the COM Registration Fix failure
+					genericTitle + " " + _("Failed"),
+					wx.OK,
+				)
+		# Display success dialog if there were no errors
 		messageBox(
 			_(
-				# Translators: The message displayed when the COM Registration Fixing tool completes.
-				"The COM Registration Fixing tool has finished. "
+				# Translators: Message shown when the COM Registration Fixing tool completes.
+				"The COM Registration Fixing Tool has completed successfully.\n"
 				"It is highly recommended that you restart your computer now, to make sure the changes take full effect.",
 			),
-			# Translators: The title of a dialog presented when the COM Registration Fixing tool is complete.
-			_("COM Registration Fixing Tool"),
+			genericTitle,
 			wx.OK,
 		)
 
+	def _CRFTIntroMessage(self) -> str:
+		"""Helper that returns the COM Registration Fixing Tool's introductory statement."""
+		return _(
+			# Translators: Explain the COM Registration Fixing tool to users before running
+			"Welcome to the COM Registration Fixing tool.\n"
+			"This tool is used by NVDA to fix problems it may have as it tries to interact with various applications, "
+			"or with Windows itself.\n"
+			"It examines the system registry for corrupted or missing accessibility entries and will correct them.\n"
+			"Those entries are sometimes damaged by installing or uninstalling programs, or other system events. This "
+			'can result in "unknown" or "pane" being spoken instead of the content you were expecting, or previously '
+			"accessible elements suddenly no longer reading correctly.\n\n"
+			"You have most likely been asked to run this tool by NVDA support or a power user trying to assist you.\n"
```

Comment:
I feel like "NVDA support" is not very well defined and "power user" might be an unfamiliar term to novice users.
I don't have a good suggestion, but perhaps something like

```suggestion
			"You have most likely been asked to run this tool by an experienced user or support worker to investigate and potentially fix an issue with NVDA's behaviour.\n"
```